cmd/genkotlin/main.go

```diff
@@ -35,15 +37,32 @@ func main() {
 		log.Fatalf("failed to load sources from %s excluding %s: %v", *inputDir, *excludeRegexpStr, err)
 	}
 
-	generator := kotlin.NewGenerator(kotlin.Config{UseJsonTagNames: *useJsonNames})
+	config := kotlin.Config{UseJsonTagNames: *useJsonNames}
+	if *outPackage != "" {
+		config.OutPackage = *outPackage
+	}
+
+	generator := kotlin.NewGenerator(config)
 	// generate kotlin classes
 	if *outputDir != "" {
 		kotlinFiles := generator.Generate(sources)
 		// save
 		for f, body := range kotlinFiles {
-			filePath := *outputDir + "/" + f + ".kt"
-			if err := ioutil.WriteFile(filePath, body, 0666); err != nil {
-				fmt.Fprintf(os.Stderr, "failed to save generated kotlin file %s: %v", filePath, err)
+			outPackagePath := ""
+			if *outPackage != "" {
+				outPackagePath = strings.Join(strings.Split(*outPackage, "."), "/")
+			}
+
+			// 0666 is enough, but https://github.com/golang/go/issues/22323
+			packagePath := *outputDir + "/" + outPackagePath
```

Comment:
pass direct path in `outputDir`, no need to concatenate path and create additional directories under the hood, it breaks tool API, when you specify output dir but actually files will be on other path which is confusing.

Reply:
So, you see it as 
`-o ~/myrepo/src/com/myapp/ -package com.myapp`
?

Reply:
Updated. Now we have to explicitly duplicate the package in the output path, as mentioned before.